client/components/bloganuary-header/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import BloganuaryIcon from 'calypso/components/blogging-prompt-card/bloganuary-icon';
+import isBloganuary from 'calypso/data/blogging-prompt/is-bloganuary';
+import { Tag } from 'calypso/reader/list-manage/types';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { successNotice } from 'calypso/state/notices/actions';
+import { requestFollowTag } from 'calypso/state/reader/tags/items/actions';
+import { getReaderFollowedTags } from 'calypso/state/reader/tags/selectors';
+import { toggleReaderSidebarTags } from 'calypso/state/reader-ui/sidebar/actions';
+import { isTagsOpen } from 'calypso/state/reader-ui/sidebar/selectors';
+import './style.scss';
+
+const containsBloganuary = ( followedTags: Tag[] | undefined ): boolean | undefined => {
+	return followedTags?.some( ( tag: Tag ) => tag.slug === 'bloganuary' );
+};
+
+const BloganuaryHeader = () => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const followedTags = useSelector( getReaderFollowedTags ) as Tag[];
+	const prevIsFollowingBloganuary = useRef( containsBloganuary( followedTags ) );
+	const isFollowingBloganuary = containsBloganuary( followedTags );
+	const isTagsSidebarOpen = useSelector( isTagsOpen );
+
+	useEffect( () => {
+		dispatch( recordTracksEvent( 'calypso_reader_bloganuary_banner_view' ) );
+	}, [ dispatch ] );
+
+	// Handle successfully subscribing to 'bloganuary' the redux-y way
+	// Errors are handled in client/state/data-layer/wpcom/read/tags/mine/new/index.js:receiveError
+	useEffect( () => {
+		// tripple equals check is needed because 'undefined' means the value has not been loaded yet.
+		if ( prevIsFollowingBloganuary.current === false && isFollowingBloganuary ) {
+			dispatch( successNotice( translate( 'Subscribed to bloganuary tag' ) ) );
+		}
+		prevIsFollowingBloganuary.current = isFollowingBloganuary;
+	}, [ isFollowingBloganuary, prevIsFollowingBloganuary, dispatch, translate ] );
```

Comment:
This does seem a bit funky to handle the success notice. I think ideally the success notice should be dispatched on the redux side when the request to follow the tag receives a valid response: that way its not implemented in each component we use the action from, but tied to the action itself.

I don't see success messages for when we follow tags from the current follow tags buttons in the tags streams or on the discover tags tabs sidebar (at least not on production or calypso.live). So I don't think we need this here specifically. But if we did, we should probably fix it for all of the follow tags buttons via the action on the redux side as mentioned above. That seems out of scope for this PR though.

Reply:
ohhh yes, you're right, I added this before I changed the button state, I originally thought we need *some* way to know that it was successful. But I think you're right, everywhere else we just change the button state.